crates/primitives/src/roles.rs

```diff
@@ -0,0 +1,57 @@
+use arbitrary::Arbitrary;
+use borsh::{BorshDeserialize, BorshSerialize};
+
+/// Roles with authority in the administration subprotocol.
+#[repr(u8)]
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    Eq,
+    PartialEq,
+    PartialOrd,
+    Ord,
+    Hash,
+    BorshSerialize,
+    BorshDeserialize,
+    Arbitrary,
+)]
+#[borsh(use_discriminant = false)]
+pub enum Role {
```

Comment:
What is going on with this derive order?  especially `Eq, PartialEq, PartialOrd, Ord` being mixed weird?

The guideline is to put the more the "essential" properties first, going to more specific ones, then the non-std ones in alphabetical order.  So *something* like: `Copy, Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Arbitrary, BorshDeserialize, BorshSerialize`.